torch/nn/_functions/rnn.py

```diff
@@ -34,9 +37,9 @@ def LSTMCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
 
 
 def GRUCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
-    gi = F.linear(input, w_ih, b_ih)
+    gi = input.unsqueeze(1).expand(input.size(0), 3, input.size(1)) if w_ih is None else F.linear(input, w_ih, b_ih)
     gh = F.linear(hidden, w_hh, b_hh)
-    i_r, i_i, i_n = gi.chunk(3, 1)
+    i_r, i_i, i_n = [x.squeeze(1) for x in gi.chunk(3, 1)]
     h_r, h_i, h_n = gh.chunk(3, 1)
```

Comment:
Would this replace the `chunk` call? Not entirely sure how this would keep the correct chunk sizes for the different gates!